prime-router/docs/design/hl7-validation/design.md

```diff
@@ -0,0 +1,115 @@
+# Problem Statement
+As a sender/receiver I want submitted/received data to be validated to ensure data integrity and quality.
+
+There needs to exist a collection of settings and reusable components to perform varying types of validation at various points in the HL7 message's lifecycle.
+
+## Proposed Design
+![hl7-validation-annotated-architecture-diagram.png](annotated-architecture-diagram.png)
+
+We propose two validation "checkpoints":
+1. Sender validation
+2. Receiver Validation
+
+This will use the setting proposed in the FHIR validation design doc called `validationProfile`. This is reusable because
+senders and receivers can only be set up to receive one format at a time, so there should not be confusion over which 
+validationProfile pertains to which format. 
+
+Each validation checkpoint shall run its configured validation and in the event of a failure: 
+- actionLog the result and error information
+- notify the sender
+
+## Background Information/Dev Notes
+### The HAPI HL7 library
+The HAPI library offers 3 "levels" of message validation:
+1. Basic Message Validation - very basic validation on message length and some segments, but the article
+(https://saravanansubramanian.com/hl72xhapivalidation/) said this will not be enough
+2. Custom Message Validation - Extends basic and enabled through the extensibility offered by the message validation 
+framework itself. Allows you to override default behaviour.
+3. Validation by the use of Conformance Profiles - enable the sharing of any stringent message specifications involving 
+specific workflows to an open registry that everyone can use. They push this one. We may be able to download something
```

Comment:
Who is they? Maybe make it more clear you are referencing the link above and provide some background on the source and why we should take its advice? It seems to be a blog by some dude, is there any guidance from an official HL7 source we can reference instead?